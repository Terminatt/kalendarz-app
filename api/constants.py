EMAIL_REGEX = r'\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,}\b'
PASSWORD_MIN_LENGHT = 9
GROUPS = ['administrator', 'regular_user']
TOKEN_EXPIRATION = 24 # In Hours