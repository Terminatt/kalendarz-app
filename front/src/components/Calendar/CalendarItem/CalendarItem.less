@import '~styles/variables.less';

.calendar-item {
  background-color: @secondary-bg;
  border: none;
  border-radius: @border-radius-calendar;
  box-shadow: @box-shadow-tiny;
  border-left: 3px solid @border-color-base;
  cursor: pointer;
  
  &-normal {
    border-color: @primary-color;
  }

  &-another-month-day {
    border-color: @calendar-day-disabled;
    color: @calendar-day-disabled;
    cursor: not-allowed;
  }

  &-before-today {
    border-color: @calendar-day-disabled;
  }

  &-today {
    border-color: @link-color;
  }

  &-dayoff {
    border-color: @red-color;
  }
}