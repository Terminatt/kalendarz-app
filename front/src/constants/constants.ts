export const formLayout = {
    labelCol: { span: 24 },
    wrapperCol: { span: 24 },
};

export const errorMessages = {
    USERNAME_TAKEN: 'Użytkownik o podanej nazwie już istnieje',
    EMAIL_TAKEN: 'Użytkownik o podanym emailu już istnieje',
    PASSWORD_TOO_SIMPLE: 'Hasło jest zbyt proste',
};
