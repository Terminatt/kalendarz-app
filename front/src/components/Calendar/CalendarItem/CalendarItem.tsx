import React, { useCallback } from 'react';

import './CalendarItem.less';

export enum CalendarItemType {
  ANOTHER_MONTH_DAY = 'another-month-day',
  BEFORE_TODAY = 'before-today',
  TODAY = 'today',
  NORMAL = 'normal',
  DAYOFF = 'dayoff',
}

export interface CalendarItemProps {
  type: CalendarItemType,
  dayNumber: string;
  dayName: string;
  onClick?: () => void;
}

const CalendarItem: React.FC<CalendarItemProps> = (props) => {
    const {
        type, dayNumber, dayName, onClick,
    } = props;
    const disabled = type === CalendarItemType.ANOTHER_MONTH_DAY || type === CalendarItemType.DAYOFF;

    const onItemClick = useCallback(() => {
        if (!onClick) {
            return;
        }
        onClick();
    }, [onClick]);

    return (
        <button onClick={onItemClick} className={`calendar-item calendar-item-${type}`} type="button" disabled={disabled}>
            <div>{dayNumber}</div>
            <div>{dayName}</div>
        </button>
    );
};

export default CalendarItem;
