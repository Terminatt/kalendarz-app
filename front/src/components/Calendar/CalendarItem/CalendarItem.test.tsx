import React from 'react';
import renderer from 'react-test-renderer';
import CalendarItem, { CalendarItemType } from './CalendarItem';
import { DAY_NAMES } from '@constants/constants';
import { fireEvent, render } from '@testing-library/react';

describe('Calendar Item Types', () => {
    beforeEach(() => {
    });
    
    it('Normal matches the snapshot', () => {
        const tree = renderer
            .create(
                <CalendarItem type={CalendarItemType.NORMAL} dayNumber={'01'} dayName={DAY_NAMES[0]} />,
            )
            .toJSON();
        
            expect(tree).toMatchSnapshot();
    });

    it('Today matches the snapshot', () => {
        const tree = renderer
            .create(
                <CalendarItem type={CalendarItemType.TODAY} dayNumber={'01'} dayName={DAY_NAMES[0]} />,
            )
            .toJSON();
        
            expect(tree).toMatchSnapshot();
    });

    it('Dayoff matches the snapshot', () => {
        const tree = renderer
            .create(
                <CalendarItem type={CalendarItemType.DAYOFF} dayNumber={'01'} dayName={DAY_NAMES[0]} />,
            )
            .toJSON();
        
            expect(tree).toMatchSnapshot();
    });

    it('Before today matches the snapshot', () => {
        const tree = renderer
            .create(
                <CalendarItem type={CalendarItemType.BEFORE_TODAY} dayNumber={'01'} dayName={DAY_NAMES[0]} />,
            )
            .toJSON();
        
            expect(tree).toMatchSnapshot();
    });

    it('Another month day today matches the snapshot', () => {
        const tree = renderer
            .create(
                <CalendarItem type={CalendarItemType.ANOTHER_MONTH_DAY} dayNumber={'01'} dayName={DAY_NAMES[0]} />,
            )
            .toJSON();
            expect(tree).toMatchSnapshot();
    });
});

describe('Calendar Item', () => {
    it('Normal day triggers click', () => {
        const onClick = jest.fn();
        const screen = render(<CalendarItem onClick={onClick} type={CalendarItemType.NORMAL} dayNumber={'01'} dayName={DAY_NAMES[0]} />)
        fireEvent.click(screen.getByText(/01/i))
        expect(onClick).toBeCalled();
    });

    it('Today day triggers click', () => {
        const onClick = jest.fn();
        const screen = render(<CalendarItem onClick={onClick} type={CalendarItemType.TODAY} dayNumber={'01'} dayName={DAY_NAMES[0]} />)
        fireEvent.click(screen.getByText(/01/i))
        expect(onClick).toBeCalled();
    });

    it('Dayoff day does not trigger click', () => {
        const onClick = jest.fn();
        const screen = render(<CalendarItem onClick={onClick} type={CalendarItemType.DAYOFF} dayNumber={'01'} dayName={DAY_NAMES[0]} />)
        fireEvent.click(screen.getByText(/01/i))
        expect(onClick).not.toBeCalled();
    });

    it('Before today day triggers click', () => {
        const onClick = jest.fn();
        const screen = render(<CalendarItem onClick={onClick} type={CalendarItemType.BEFORE_TODAY} dayNumber={'01'} dayName={DAY_NAMES[0]} />)
        fireEvent.click(screen.getByText(/01/i))
        expect(onClick).toBeCalled();
    });

    it('Another month day does not trigger click', () => {
        const onClick = jest.fn();
        const screen = render(<CalendarItem onClick={onClick} type={CalendarItemType.ANOTHER_MONTH_DAY} dayNumber={'01'} dayName={DAY_NAMES[0]} />)
        fireEvent.click(screen.getByText(/01/i))
        expect(onClick).not.toBeCalled();
    });
})