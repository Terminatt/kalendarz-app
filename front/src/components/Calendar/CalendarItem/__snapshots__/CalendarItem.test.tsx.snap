// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar Item Types Another month day today matches the snapshot 1`] = `
<button
  className="calendar-item calendar-item-another-month-day"
  disabled={true}
  onClick={[Function]}
  type="button"
>
  <div>
    01
  </div>
  <div>
    Nd
  </div>
</button>
`;

exports[`Calendar Item Types Before today matches the snapshot 1`] = `
<button
  className="calendar-item calendar-item-before-today"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <div>
    01
  </div>
  <div>
    Nd
  </div>
</button>
`;

exports[`Calendar Item Types Dayoff matches the snapshot 1`] = `
<button
  className="calendar-item calendar-item-dayoff"
  disabled={true}
  onClick={[Function]}
  type="button"
>
  <div>
    01
  </div>
  <div>
    Nd
  </div>
</button>
`;

exports[`Calendar Item Types Normal matches the snapshot 1`] = `
<button
  className="calendar-item calendar-item-normal"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <div>
    01
  </div>
  <div>
    Nd
  </div>
</button>
`;

exports[`Calendar Item Types Today matches the snapshot 1`] = `
<button
  className="calendar-item calendar-item-today"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <div>
    01
  </div>
  <div>
    Nd
  </div>
</button>
`;
